ykpack/src/types.rs

```diff
@@ -225,21 +227,85 @@ impl Display for Operand {
 
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
 pub enum Constant {
-    UnsignedInt(UnsignedInt),
-    SignedInt(SignedInt),
+    Int(ConstantInt),
     Unimplemented,
 }
 
 impl Display for Constant {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            Constant::UnsignedInt(u) => write!(f, "{}", u),
-            Constant::SignedInt(s) => write!(f, "{}", s),
+            Constant::Int(i) => write!(f, "{}", i),
             Constant::Unimplemented => write!(f, "Unimplemented Constant"),
         }
     }
 }
 
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub enum ConstantInt {
+    UnsignedInt(UnsignedInt),
+    SignedInt(SignedInt),
+}
+
+impl ConstantInt {
+    // When constructing constant integers, truncation is deliberate.
```

Comment:
What does this mean?

Reply:
You might be able to do something like:

```rust
impl From<uwhatever> for ConstantInt {
    fn from(x: uwhatever) -> Self { ... }
}
```

and then avoid having to call differently named functions.

Reply:
> What does this mean?

It's justification for the excessive use of `as`.

> You might be able to do something like:

Can I special-case u128? It breaks the mould sadly.

Reply:
> Can I special-case u128? It breaks the mould sadly.

I don't think it's going to work. Each different type lowers to a different ykpack variant, so each type needs different functionality anyway.

Reply:
>Each different type lowers to a different ykpack variant, so each type needs different functionality anyway.

OK.

Can we make the comment a bit less gnomic?

Reply:
Let me try some macros.